cas-server-core-util/src/main/java/org/jasig/cas/util/DateTimeUtils.java

```diff
@@ -0,0 +1,152 @@
+package org.jasig.cas.util;
+
+import org.joda.time.DateTime;
+import org.joda.time.ReadableInstant;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.chrono.ChronoZonedDateTime;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * @author tduehr
+ * @since 4.3.0
+ */
+public final class DateTimeUtils {
+    /**
+     * Unix Epoch 1/1/70 UTC.
+     */
+    public static final ZonedDateTime EPOCH = ZonedDateTime.ofInstant(Instant.EPOCH, ZoneOffset.UTC);
+
+    private DateTimeUtils() {
+    }
+
+    /**
+     * Convenience method to get a ZonedDateTime instance of Instant.EPOCH.
+     * @return ZonedDateTime representing Epoch.
+     */
+    public static ZonedDateTime getEpoch() {
+        return EPOCH;
+    }
+
+    /**
+     * Utility for creating a ZonedDateTime object from a millisecond timestamp.
+     * @param time Miliseconds since Epoch UTC
+     * @return ZonedDateTime representing time
+     */
+    public static ZonedDateTime of(final long time){
+        return of(time, ZoneOffset.UTC);
+    }
+
+    /**
+     * Utility for creating a ZonedDateTime object from a millisecond timestamp.
+     * @param time Miliseconds since Epoch
+     * @param zoneId Time zone
+     * @return ZonedDateTime representing time
+     */
+    public static ZonedDateTime of(final long time, final ZoneId zoneId){
+        return ZonedDateTime.ofInstant(Instant.ofEpochMilli(time), zoneId);
+    }
+
+    /**
+     * Gets ZonedDateTime for ReadableInstant.
+     * @param time Time object to be converted.
+     * @return ZonedDateTime representing time
+     * @deprecated Java 8 transition utillity method
```

Comment:
I understand the rationale behind the this move; I do think we ought to remove them completely. If they should not be used, they should not be there at all since it doesn't make much sense to create a brand new class and mark methods inside it as deprecated, specially when no other methods or components access the deprecated functionality. If you were changing an existing class, I'd have been in favor deprecating things.